wheel/pep425tags.py

```diff
@@ -105,17 +109,85 @@ def get_abi_tag():
     return abi
 
 
-def get_platform():
+def calculate_macosx_platform_tag(archive_root, platform_tag):
+    """
+    Calculate proper macosx platform tag basing on files which are included to wheel
+    """
+    prefix, base_version, suffix = platform_tag.split('-')
+    base_version = tuple([int(x) for x in base_version.split(".")])
+    if len(base_version) == 2:
+        base_version = base_version + (0,)
+
+    assert len(base_version) == 3
+    if "MACOSX_DEPLOYMENT_TARGET" in os.environ:
+        deploy_target = tuple([int(x) for x in os.environ[
+            "MACOSX_DEPLOYMENT_TARGET"].split(".")])
+        if len(deploy_target) == 2:
+            deploy_target = deploy_target + (0,)
+        if deploy_target < base_version:
+            sys.stderr.write(
+                "[WARNING] MACOSX_DEPLOYMENT_TARGET is set "
+                "to lower value than your python is compiled\n"
+                )
+        else:
+            base_version = deploy_target
+
+    assert len(base_version) == 3
+    start_version = base_version
+    versions_dict = {}
+    for (dirpath, dirnames, filenames) in os.walk(archive_root):
+        for filename in filenames:
+            if filename.endswith('.dynlib') or filename.endswith('.so'):
+                lib_path = os.path.join(dirpath, filename)
+                versions_dict[lib_path] = extract_macosx_min_system_version(lib_path)
+
+    if len(versions_dict) > 0:
+        base_version = max(base_version, max(versions_dict.values()))
+
+    if base_version[-1] == 0:
+        fin_base_version = base_version[:-1]
+    else:
+        fin_base_version = base_version
+
+    fin_base_version = "_".join([str(x) for x in fin_base_version])
+    if start_version < base_version:
+        problematic_files = [k for k, v in versions_dict.items() if v > start_version]
+        problematic_files = "\n".join(problematic_files)
+        error_message = \
+            "[WARNING] This wheel needs higher macosx version than {} " \
+            "is set in MACOSX_DEPLOYMENT_TARGET variable. " \
```

Comment:
I do not understand. I use this text in format.

Reply:
Depending on `if "MACOSX_DEPLOYMENT_TARGET" in os.environ`, it can be that `"is set in MACOSX_DEPLOYMENT_TARGET variable."` will be twice in the string, since you format it later and it is already in the format string.